provider/oracle/network/environ.go

```diff
@@ -0,0 +1,348 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/go-oracle-cloud/api"
+	"github.com/juju/go-oracle-cloud/common"
+	"github.com/juju/go-oracle-cloud/response"
+	"github.com/juju/loggo"
+	names "gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/instance"
+	"github.com/juju/juju/network"
+	commonProvider "github.com/juju/juju/provider/oracle/common"
+)
+
+var logger = loggo.GetLogger("juju.provider.oracle.network")
+
+type NetworkingAPI interface {
+	commonProvider.Instancer
+	commonProvider.Composer
+
+	AllIpNetworks([]api.Filter) (response.AllIpNetworks, error)
+}
+
+// Environ type that implement the Networking interface
+// from the environ package.
+// This defines methods that the oracle environment
+// will use to support netowrking capabilities
+type Environ struct {
+	client NetworkingAPI
+}
+
+var _ environs.Networking = (*Environ)(nil)
+
+// NewEnviron returns a new network Environment that has
+// network capabilities inside the oracle cloud environment
+func NewEnviron(api NetworkingAPI) *Environ {
+	return &Environ{
+		client: api,
+	}
+}
+
+// Subnets returns basic information about subnets known by the provider
+// for the oracle cloud environment
+func (e Environ) Subnets(
+	id instance.Id,
+	subnets []network.Id,
+) ([]network.SubnetInfo, error) {
+
+	ret := []network.SubnetInfo{}
+	found := make(map[string]bool)
```

Comment:
There is a package "github.com/juju/utils/set" which makes handling of 'found' easier.